File: 1]. Data Structures/01]. Array/cpp/03_containsDuplicate.cpp
#include <bits/stdc++.h>
using namespace std;

void solve() {
    int n;
    cin >> n;
    
    vector<int> nums(n);
    for(int i = 0; i < n; i++) {
        cin >> nums[i];
    }

    unordered_set<int> seen;
    bool has_duplicate = false;
    
    for (int i = 0; i < n; i++) {
        if (seen.count(nums[i])) {
            has_duplicate = true;
            break; 
        }
        seen.insert(nums[i]);
    }
    
    if (has_duplicate) {
        cout << "true\n";
    } else {
        cout << "false\n";
    }
}

int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);
    
    int t;
    if (cin >> t) {
        while (t--) {
            solve();
        }
    }
    
    return 0;
}

File: 1]. Data Structures/01]. Array/python/03_containsDuplicate.py
import sys

def solve():
    n_str = sys.stdin.readline().strip()
    if not n_str:
        return
    n = int(n_str)
    
    if n == 0:
        print("false")
        return
        
    nums = list(map(int, sys.stdin.readline().split()))
    
    seen = set()
    has_duplicate = False
    
    for i in range(n):
        if nums[i] in seen:
            has_duplicate = True
            break
        seen.add(nums[i])
        
    if has_duplicate:
        print("true")
    else:
        print("false")

if __name__ == '__main__':
    t_str = sys.stdin.readline().strip()
    if t_str:
        t = int(t_str)
        for _ in range(t):
            solve()